### DISCUSSION TITLE
DB().NewQuery().All() does not populate Record.collection?

### DISCUSSION THREAD

**Discussion by Gys** [pocketbase/pocketbase#5884] [Q&A] [answered]
I have a rather complicated query with some sub queries and therefore want (need?) to use DB().NewQuery(). The resulting records should go into an existing table named Locations.

This works as expected:
```
locs := []*models.Record{}
err := pb.Dao().RecordQuery("locations").Limit(10).All(&locs)
```
The locs slice ends up with all BaseModel data and all custom fields are saved in the .collection field.

But this does not work in the same way:
```
locs := []*models.Record{}
err := pb.Dao().DB().NewQuery(`SELECT locations.* FROM locations LIMIT 10`).All(&locs)
```
Now the locs slice only holds the BaseModel data but the .collection is nil.

How can I fill the .collection as well when using a NewQuery()?

**Comment by ganigeorgiev** [comment]
Technically it is possible but I don't feel confident recommending it because it is no longer allowed in v0.23.0.

In general with <= v0.22.x you can do something like:

```go
rows := []dbx.NullStringMap{}
err := pb.Dao().DB().NewQuery("").All(&rows)

locations, err := pb.Dao().FindCollectionByNameOrId("locations")
records := models.NewRecordsFromNullStringMaps(locations, rows)
```

But again, the above will no longer be allowed in v0.23.0 because the records scanning is a lot more involving.

It depends on your actual query, but one possible alternative is to continue using `RecordQuery()` and use the `dbx.NewExp` with your raw statement(s), e.g.:

```go
locs := []*models.Record{}
err := pb.Dao().RecordQuery("locations").AndWhere(dbx.NewExp("anything ...")).Limit(10).All(&locs)
```

**Comment by Gys** [comment, marked as answer]
My subqueries are in the limit and offset 😅

Edit: but it seems the limit and offset can also be defined inside the .Where() so I try to work with that 👍
